src/main/scala/beam/router/graphhopper/AbstractBeamTimeDetails.scala

```diff
@@ -0,0 +1,39 @@
+package beam.router.graphhopper
+
+import com.graphhopper.routing.weighting.Weighting
+import com.graphhopper.util.details.AbstractPathDetailsBuilder
+import com.graphhopper.util.{EdgeIteratorState, GHUtility}
+
+import scala.util.Try
+
+/**
+  * It's custom implementation of TimeDetails
+  */
+abstract class AbstractBeamTimeDetails(val weighting: Weighting, name: String, val reverse: Boolean)
+    extends AbstractPathDetailsBuilder(name) {
+
+  private var prevEdgeId = -1
+  // will include the turn time penalty
+  private var time = 0L
+
+  override def isEdgeDifferentToLastEdge(edge: EdgeIteratorState): Boolean = {
+    if (edge.getEdge != prevEdgeId) {
+      time = weighting match {
+        case bWeighting: BeamWeighting =>
+          Try(bWeighting.calcEdgeMillisForDetails(edge, reverse))
```

Comment:
What about possible exception (it is wrapped by `Try`), should it be logged?

Reply:
This `Try` to avoid some inner GH exception. GH calculate routing with 2 iterations. One of them with global flag `reverse=true`. Other one - `reverse=false`. With this `Try` trying guess this value.